firebase-token-generator/firebase-token-generator.d.ts

```diff
@@ -0,0 +1,49 @@
+// Type definitions for firebase-token-generator v2.0.0
+// Project: https://github.com/firebase/firebase-token-generator-node
+// Definitions by: Hans Van den Keybus <https://github.com/dotdotcommadot>
+// Definitions: https://github.com/borisyankov/DefinitelyTyped
+
+interface TokenOptions {
+  expires?: number;
+  notBefore?: number;
+  admin?: boolean;
+  debug?: boolean;
+  simulate?: boolean;
+  iat?: number;
+}
+
+interface FirebaseTokenGenerator {
+
+  /**
+   * Builds a new object that can generate Firebase authentication tokens.
+   * @constructor
+   * @param { String } secret The secret for the Firebase being used (get yours from the Firebase Admin Console).
+   */
+  new (secret: string): FirebaseTokenGenerator;
```

Comment:
This looks suspicious - the primary object when you import this module will be a `FirebaseTokenGenerator`, which you can `new` up to get...a `FirebaseTokenGenerator`? That means this will work:

```ts
import ftg = require("firebase-token-generator");

let x = new new new new new new new new ftg("")("")("")("")("")("")("")("");
```

I think what you want to do is convert this interface into a class, and convert this into the constructor.